kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
I'm not sure I'm following now

>I think we want that to be an or rather than requiring the flag. 

Are you suggesting that we keep the ready flag, but switch to "ready || (lastModForTarget > lastModForSource)"?

>What if the data was written by something other than Kite and then consumed? There's no reason not to use it if its modified time is more recent.

I think if something was written into a Kite target outside of Kite (highly dangerous I would think), or that something was written into the Kite view target (using Kite) but outside of MapReduce that the output would be highly suspicious. It seems like things from the input would have been missed, or else it would have been run through M/R in the first place. Usage for WriteMode.CHECKPOINT is usually that you're sure you'll achieve the same output by running the stage as you already have. The best way to ensure that is that the output was achieve by running through M/R (in which case the ready flag is present) and that the source is older than the target, which isn't necessarily guaranteed by the ready flag.